app/src/main/java/net/squanchy/notification/NotificationsIntentService.java

```diff
@@ -0,0 +1,82 @@
+package net.squanchy.notification;
+
+import android.app.AlarmManager;
+import android.app.IntentService;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import net.squanchy.R;
+import net.squanchy.schedule.domain.view.Event;
+
+import org.joda.time.LocalDateTime;
+
+import io.reactivex.Observable;
+
+public class NotificationsIntentService extends IntentService {
+
+    private static final int NOTIFICATION_INTERVAL_MINUTES = 10;
+    private static final boolean SHOW_NOTIFICATIONS_DEFAULT = true;
+
+    private NotificationService service;
+    private NotificationCreator notificationCreator;
+    private Notifier notifier;
+    private SharedPreferences preferences;
+
+    public NotificationsIntentService() {
+        super(NotificationsIntentService.class.getSimpleName());
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        NotificationComponent component = NotificationInjector.obtain(this);
+        service = component.service();
+        notificationCreator = new NotificationCreator(this);
+        notifier = Notifier.from(this);
+        preferences = PreferenceManager.getDefaultSharedPreferences(this);
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+        LocalDateTime notificationIntervalEnd = new LocalDateTime().plusMinutes(NOTIFICATION_INTERVAL_MINUTES);
+
+        Observable<Event> sortedFavourites = service.sortedFavourites();
+
+        LocalDateTime now = new LocalDateTime();
+        if (shouldShowNotifications()) {
+            sortedFavourites
+                    .filter(event -> event.startTime().isAfter(now))
```

Comment:
Rx operators are to be used for streams, here the single atomic event should be a list of events, not a single event, so the transformation should be from list to list more than from event to event.